Mathlib/CategoryTheory/Countable.lean

```diff
@@ -0,0 +1,71 @@
+/-
+Copyright (c) 2023 Dagur Asgeirsson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Dagur Asgeirsson
+-/
+import Mathlib.Data.Countable.Small
+import Mathlib.CategoryTheory.FinCategory
+/-!
+# Countable categories
+
+A category is countable in this sense if it has countably many objects and countably many morphisms.
+
+-/
+
+universe w v u
+
+open Classical
+
+noncomputable section
+
+namespace CategoryTheory
+
+instance discreteCountable {α : Type*} [Countable α] : Countable (Discrete α) :=
+  Countable.of_equiv α discreteEquiv.symm
+
+/-- A category with countably many objects and morphisms. -/
+class CountableCategory (J : Type*) [Category J] : Prop where
+  countableObj : Countable J := by infer_instance
+  countableHom : ∀ j j' : J, Countable (j ⟶ j') := by infer_instance
+
+attribute [instance] CountableCategory.countableObj CountableCategory.countableHom
+
+instance countablerCategoryDiscreteOfCountable (J : Type*) [Countable J] :
+    CountableCategory (Discrete J) where
+
+namespace CountableCategory
+
+variable (α : Type*) [Countable α] [Category α] [CountableCategory α]
+
+/-- A countable category `α` is equivalent to a category with objects in `Type`. -/
+abbrev ObjAsType : Type :=
+  InducedCategory α (equivShrink.{0} α).symm
+
+instance {i j : ObjAsType α} : Countable (i ⟶ j) :=
+  CountableCategory.countableHom ((equivShrink.{0} α).symm i) _
+
+/-- The constructed category is indeed equivalent to `α`. -/
+noncomputable def objAsTypeEquiv : ObjAsType α ≌ α :=
```

Comment:
I think you would need to shrink the universes of the arrows as well?

Reply:
I was just copying the `FinCategory` file here, I'm not using this for anything. 
But indeed it is also equivalent to a category with both objects and morphisms in `Type`. I don't need that fact either.

Reply:
Ok I used this to do the universe polymorphic version of `HasCountableLimits.out` that you asked for. But I didn't need to shrink the universes of the arrows.